File: Solutions/4. Best Time to Buy and Sell Stock/sol.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        mxprofit = 0
        minpurchase = prices[0]
        
        for i in range(1 , len(prices)):
            mxprofit = max(mxprofit , prices[i] - minpurchase)
            minpurchase = min(prices[i] , minpurchase)
                
        return mxprofit

File: Solutions/4. Best Time to Buy and Sell Stock/sol.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mx = 0 , mn = prices[0] ;
        for(int i = 1 ; i < prices.size() ; i++)
        {
            mx = max(mx , prices[i] - mn) ;
            mn = min(prices[i] , mn) ;
        }
        return mx ;
    }
};